test/bin.spec.js

```diff
@@ -125,3 +136,87 @@ test('w3 ls', async (t) => {
   const list1 = await execa('./bin.js', ['ls'], { env })
   t.notThrows(() => CID.parse(list1.stdout.trim()))
 })
+
+test('w3 delegation create', async t => {
+  const { stdout } = await execa('./bin.js', ['space', 'create'], { env: createEnv() })
+  const spaceDID = DID.parse(stdout.trim()).did()
+
+  const bob = await Signer.generate()
+  const proofPath = path.join(os.tmpdir(), `w3cli-test-delegation-${Date.now()}`)
+
+  await execa('./bin.js', ['delegation', 'create', bob.did(), '--output', proofPath], { env: createEnv() })
+
+  const reader = await CarReader.fromIterable(fs.createReadStream(proofPath))
+  const blocks = []
+  for await (const block of reader.blocks()) {
+    blocks.push(block)
+  }
+
+  // @ts-expect-error
+  const delegation = importDAG(blocks)
+  t.is(delegation.audience.did(), bob.did())
+  t.is(delegation.capabilities[0].can, '*')
+  t.is(delegation.capabilities[0].with, spaceDID)
+})
+
+test('w3 space add', async t => {
+  const aliceEnv = createEnv()
+  const bobEnv = createEnv({ storeName: 'w3cli-test-bob' })
+
+  const aliceOut0 = await execa('./bin.js', ['space', 'create'], { env: aliceEnv })
+  const spaceDID = DID.parse(aliceOut0.stdout.trim()).did()
+
+  const bobOut0 = await execa('./bin.js', ['whoami'], { env: bobEnv })
+  const bobDID = DID.parse(bobOut0.stdout.trim()).did()
+
+  const proofPath = path.join(os.tmpdir(), `w3cli-test-delegation-${Date.now()}`)
+
+  await execa('./bin.js', ['delegation', 'create', bobDID, '--output', proofPath], { env: aliceEnv })
+
+  const bobOut1 = await execa('./bin.js', ['space', 'ls'], { env: bobEnv })
+  t.false(bobOut1.stdout.includes(spaceDID))
+
+  const bobOut2 = await execa('./bin.js', ['space', 'add', proofPath], { env: bobEnv })
+  t.is(bobOut2.stdout.trim(), spaceDID)
+
+  const bobOut3 = await execa('./bin.js', ['space', 'ls'], { env: bobEnv })
+  t.true(bobOut3.stdout.includes(spaceDID))
+})
+
+test('w3 space ls', async t => {
+  const aliceOut0 = await execa('./bin.js', ['space', 'ls'], { env: createEnv() })
+
+  const aliceOut1 = await execa('./bin.js', ['space', 'create'], { env: createEnv() })
+  const spaceDID = DID.parse(aliceOut1.stdout.trim()).did()
+
+  const aliceOut2 = await execa('./bin.js', ['space', 'ls'], { env: createEnv() })
+
+  t.false(aliceOut0.stdout.includes(spaceDID))
+  t.true(aliceOut2.stdout.includes(spaceDID))
+})
+
+test('w3 space use', async t => {
```

Comment:
needs test for 
- `w3 space use <valid DID but not a space we created here>`
- `w3 space use <not a DID and not a name for a space we know>`

Do we have any limitations on what chars a space name can include?

Reply:
https://github.com/web3-storage/w3cli/issues/16